multiqc/modules/dragen/coverage_per_contig.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import itertools
+import math
+import re
+from collections import OrderedDict, defaultdict
+from multiqc.modules.base_module import BaseMultiqcModule
+from multiqc.plots import linegraph, bargraph
+
+# Initialise the logger
+import logging
+log = logging.getLogger(__name__)
+
+
+class DragenCoveragePerContig(BaseMultiqcModule):
+    def add_coverage_per_contig(self):
+        perchrom_data_by_phenotype_by_sample = defaultdict(dict)
+
+        for f in self.find_log_files('dragen/wgs_contig_mean_cov'):
+            perchrom_data_by_phenotype = parse_wgs_contig_mean_cov(f)
+            if f['s_name'] in perchrom_data_by_phenotype_by_sample:
+                log.debug('Duplicate sample name found! Overwriting: {}'.format(f['s_name']))
+            self.add_data_source(f, section='stats')
+            perchrom_data_by_phenotype_by_sample[f['s_name']].update(perchrom_data_by_phenotype)
+
+        # Filter to strip out ignored sample names:
+        perchrom_data_by_phenotype_by_sample = self.ignore_samples(perchrom_data_by_phenotype_by_sample)
+
+        # Merge tumor and normal data:
+        perchrom_data_by_sample = defaultdict(dict)
+        for sn in perchrom_data_by_phenotype_by_sample:
+            for phenotype in perchrom_data_by_phenotype_by_sample[sn]:
+                new_sn = sn
+                if phenotype == 'normal':
+                    new_sn = sn + '_normal'
+                perchrom_data_by_sample[new_sn] = perchrom_data_by_phenotype_by_sample[sn][phenotype]
+
+        if not perchrom_data_by_sample:
+            return
+        log.info('Found DRAGEN per-contig coverage histogram for {} DRAGEN output prefixes'.format(
+            len(perchrom_data_by_sample)))
+
+        main_contigs_by_sample = {sn: data[0] for sn, data in perchrom_data_by_sample.items()}
+        other_contigs_by_sample = {sn: data[1] for sn, data in perchrom_data_by_sample.items()}
+
+        self.add_section(
+            name='Coverage per contig',
+            anchor='dragen-coverage-per-contig',
+            description='Average coverage per contig or chromosome. '
+                        'Calculated as the number of bases (excluding duplicate marked reads, reads '
+                        'with MAPQ=0, and clipped bases), divided by the length of the contig or '
+                        '(if a target bed is used) the total length of the target region spanning that contig.',
+            plot=linegraph.plot(main_contigs_by_sample, pconfig={
+                 'id': 'dragen_coverage_per_contig',
+                 'title': 'Average coverage per contig' + (' (main contigs)' if other_contigs_by_sample else ''),
+                 'ylab': 'Average coverage',
+                 'xlab': 'Region',
+                 'categories': True,
+                 'tt_label': '<b>{point.x}</b>: {point.y:.1f}x',
+            }) + (
+                '<br>' +
+                'Non-main contigs: '
```

Comment:
I think it might be nicer to have these as separate report sections? I didn't realise that they weren't at first. I think that they're different enough to warrant it, and makes the code cleaner and report navigation easier.

Reply:
Not even sure now why I did it this way. Changed into 2 separate sections now